app/uk/gov/hmrc/economiccrimelevyregistration/models/AmlSupervisorType.scala

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.models
+
+import play.api.data.FormError
+import play.api.i18n.Messages
+import play.api.libs.json._
+import uk.gov.hmrc.economiccrimelevyregistration.config.AppConfig
+import uk.gov.hmrc.govukfrontend.views.Implicits.RichSelect
+import uk.gov.hmrc.govukfrontend.views.html.components.GovukSelect
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.Text
+import uk.gov.hmrc.govukfrontend.views.viewmodels.errormessage.ErrorMessage
+import uk.gov.hmrc.govukfrontend.views.viewmodels.label.Label
+import uk.gov.hmrc.govukfrontend.views.viewmodels.radios.RadioItem
+import uk.gov.hmrc.govukfrontend.views.viewmodels.select.{Select, SelectItem}
+import uk.gov.hmrc.hmrcfrontend.views.viewmodels.accessibleautocomplete.AccessibleAutocomplete
+
+sealed trait AmlSupervisorType
+
+case object Hmrc extends AmlSupervisorType
+case object GamblingCommission extends AmlSupervisorType
+case object FinancialConductAuthority extends AmlSupervisorType
+case object Other extends AmlSupervisorType
+
+object AmlSupervisorType {
+
+  val values: Seq[AmlSupervisorType] = Seq(
+    Hmrc,
+    GamblingCommission,
+    FinancialConductAuthority,
+    Other
+  )
+
+  def options(appConfig: AppConfig, govukSelect: GovukSelect, formErrors: Seq[FormError])(implicit
+    messages: Messages
+  ): Seq[RadioItem] = {
+    val amlProfessionalBodySupervisorOptions: Seq[SelectItem] =
+      SelectItem(
+        text = ""
+      ) +: appConfig.amlProfessionalBodySupervisors.map { opb =>
+        SelectItem(
+          value = Some(opb),
+          text = messages(s"amlSupervisor.opb.$opb")
+        )
+      }
+
+    val selectAmlProfessionalBody = govukSelect(
+      Select(
+        id = "otherProfessionalBody",
+        name = "otherProfessionalBody",
+        items = amlProfessionalBodySupervisorOptions,
+        label = Label(
+          content = Text(messages("amlSupervisor.selectFromList"))
+        ),
+        errorMessage = if (formErrors.exists(_.key == "otherProfessionalBody")) {
+          Some(ErrorMessage(content = Text(messages("amlSupervisor.selectFromList"))))
+        } else { None }
+      ).asAccessibleAutocomplete(
+        Some(AccessibleAutocomplete(defaultValue = Some(""), showAllValues = true, autoSelect = true))
+      )
+    )
+
+    values.zipWithIndex.map { case (value, index) =>
+      value match {
+        case Other =>
+          RadioItem(
+            content = Text(messages(s"amlSupervisor.${value.toString}")),
+            value = Some(value.toString),
+            id = Some(s"value_$index"),
+            conditionalHtml = Some(selectAmlProfessionalBody)
+          )
+        case _     =>
+          RadioItem(
+            content = Text(messages(s"amlSupervisor.${value.toString}")),
+            value = Some(value.toString),
+            id = Some(s"value_$index")
+          )
```

Comment:
think we can shorten this by just putting a if statement where we assign the value to `conditionalHtml` e.g.

`if(value == Other) Some(selectAmlProfessionalBody) else None`